yaml.qmd

```diff
@@ -0,0 +1,874 @@
+# Setting options with YAML {#sec-yaml}
+
+## Overview 
+
+There's a lot going on when you set a Quarto option.
+You need to:
+
+-   know the name of the option and the values it can take,
+-   figure out where to set the option, and
+-   get the YAML syntax right for setting the option to its value.
+
+In this chapter ...
+
+Options for Quarto projects, documents and code cells are set using a language called [YAML](https://yaml.org).
+YAML describes the syntax of setting options, but the options that are available and the values they can take are defined by Quarto.
+
+By now you've seen Quarto options set using YAML syntax in a number of places:
+
+-   in the project configuration file `_quarto.yml`
+-   in the document header, i.e. between the `---` at the top of a Quarto document
+-   in code cells, on lines prefaced by `#|`
+
+You'll also learn about one new place you can set options, `_metadata.yml`.
+
+## General YAML syntax
+
+In this section you'll learn the necessary YAML fundamentals to avoid syntax errors when setting options.
+Although the syntax applies generally to any YAML, all the examples you'll see are valid Quarto options and values.
+
+### YAML Mappings
+
+A YAML mapping is the specification of a key-value pair using the syntax `key: value`.
+Values can be simple, like a string, boolean, or number:
+
+``` {.yaml filename="document.qmd"}
+---
+title: My Document  # A string
+toc: true           # A boolean, `true`/`false`
+toc-depth: 2        # A number
```

Comment:
Should we use code annotation for these instead of comments? Is that even possible?

Reply:
Yeah, that might be nice. I'll do it here. But, when I repeat this example later, I'm going to use comments, so I can still have a tip about what YAML comments look like 😆 .